lib/srfi/99/records/syntactic.scm

```diff
@@ -10,11 +10,13 @@
             (procs (cddr expr))
             (make (car procs))
             (make-name (if (eq? make #t)
-                           (string->symbol (string-append "make-" name-str))
+                           (datum->syntax (car expr)
+					  (string->symbol (string-append "make-" name-str)))
```

Comment:
I wonder `(car expr)` should be `(cadr expr)` for the compatibility to SRFI 99 reference implementation.
This changes the macro expansion result if define-record-type and record-name are inserted separately.

Reply:
Thanks for the comment.

The spec does not seem to be totally clear about it. I haven't taken a look at SRFI 99's reference implementation, so `(car expr)' was my reading of the spec. However, `(cadr expr)' makes at least as much sense and we shouldn't needlessly deviate from the reference implementation.

R6RS has a similar feature where the constructor and predicate name are derived from the record name. I have taken a look at Chez's implementation. It is also consistent with SRFI 99's reference implementation in that the lexical context is taken from the record name identifier and not from the lexical context of the macro call.